tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs

```diff
@@ -0,0 +1,289 @@
+                        .ToDictionary(path => path.Path);
+
+            var changedConfigurationsByProduct = GetChangedConfigurations()
+               .GroupBy(config => config.Path.Product);
+
+            foreach (var productGroup in changedConfigurationsByProduct)
+            {
+                var configurations = productGroup.ToList();
+                var product = _products[productGroup.Key];
+
+                try
+                {
+                    product.AssignConfigs(configurations);
+                    CacheAppliedConfigurations(product, configurations);
+                }
+                catch (Exception e)
+                {
+                    Log.Warning($"Failed to apply remote configurations {e.Message}");
+                }
+            }
+
+            UncacheRemovedConfigurations();
+
+            IEnumerable<RemoteConfiguration> GetChangedConfigurations()
+            {
+                var signed = response.Targets.Signed.Targets;
+                var targetFiles = (response.TargetFiles ?? Enumerable.Empty<RcmFile>()).ToDictionary(f => f.Path, f => f);
+
+                foreach (var kp in actualConfigPath)
+                {
+                    if (!signed.TryGetValue(kp.Key, out var signedTarget))
+                    {
+                        throw new RemoteConfigurationException($"Missing config {kp.Key} in targets");
+                    }
+
+                    if (!_products.TryGetValue(kp.Value.Product, out var product))
+                    {
+                        throw new RemoteConfigurationException($"Received config {kp.Key} for a product that was not requested");
+                    }
+
+                    var isConfigApplied = product.AppliedConfigurations.TryGetValue(kp.Key, out var appliedConfig) && appliedConfig.Hashes.SequenceEqual(signedTarget.Hashes);
+                    if (isConfigApplied)
+                    {
+                        continue;
+                    }
+
+                    yield return new RemoteConfiguration(kp.Value, targetFiles[kp.Key].Raw, signedTarget.Length, signedTarget.Hashes, signedTarget.Custom.V);
+                }
+            }
+
+            void CacheAppliedConfigurations(Product product, List<RemoteConfiguration> configurations)
+            {
+                foreach (var config in configurations)
+                {
+                    var remoteConfigurationCache = new RemoteConfigurationCache(config.Path, config.Length, config.Hashes, config.Version);
+
+                    if (product.AppliedConfigurations.ContainsKey(config.Path.Path))
+                    {
+                        product.AppliedConfigurations[config.Path.Path] = remoteConfigurationCache;
+                    }
+                    else
+                    {
+                        product.AppliedConfigurations.Add(config.Path.Path, remoteConfigurationCache);
+                    }
+                }
+            }
+
+            void UncacheRemovedConfigurations()
```

Comment:
```suggestion
            void UnapplyRemovedConfigurations()
```
Just a thought - for consistency with terminology used in the RFC.

Reply:
Done at 10ea17673b64235f948ebde1f87ba0bee0454791